streampipes-client-go/streampipes/data_lake_measure_api.go

```diff
@@ -0,0 +1,117 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package streampipes
+
+import (
+	"github.com/apache/streampipes/streampipes-client-go/streampipes/config"
+	"github.com/apache/streampipes/streampipes-client-go/streampipes/internal/serializer"
+	"github.com/apache/streampipes/streampipes-client-go/streampipes/internal/streampipes_http"
+	"github.com/apache/streampipes/streampipes-client-go/streampipes/model/data_lake"
+	"log"
+)
+
+/*
+DataLakeMeasure connects to the DataLakeMeasure endpoint of Streampipes.
+DataLakeMeasure supports GET and DELETE to delete or obtain resources
+The specific interaction behavior is provided by the method bound to the DataLakeMeasure struct.
+*/
+
+type DataLakeMeasure struct {
+	config      config.StreamPipesClientConnectionConfig
+	httpRequest streampipes_http.HttpRequest
+}
+
+func NewDataLakeMeasures(clientConfig config.StreamPipesClientConnectionConfig) *DataLakeMeasure {
+	//NewDataLakeMeasure is used to return an instance of *DataLakeMeasure,
+
+	return &DataLakeMeasure{
+		config:      clientConfig,
+		httpRequest: nil,
+	}
+}
+
+func (d *DataLakeMeasure) AllDataLakeMeasure() []data_lake.DataLakeMeasure {
+	//Get a list of all measure
+
+	d.httpRequest = &streampipes_http.GetRequest{
+		HttpRequest:  streampipes_http.NewHttpRequest(d.config),
+		Deserializer: &serializer.UnmarshalDataLakeMeasure{},
+	}
+	d.resourcePath([]string{"measurements"})
```

Comment:
This httpRequest has methods like GET and DELETE, and if combine them by defining a new method that passes httpRequest as a parameter, it seems like it can only be the interface type.  I think maybe we don't need to merge this?

Reply:
What's the disadvantage of passing it as an interface type? The lifecycle of the httpRequest is limited to each query method. It doesn't make sense to make it as a field of the struct.